app/assets/javascripts/components/buttons/generic-button-component.js

```diff
@@ -0,0 +1,41 @@
+ManageIQ.angular.app.component('genericButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    onSaveClicked: '&?',
+    onAddClicked: '&?',
+  },
+  controllerAs: 'vm',
+  controller: genericButtonController,
+  templateUrl: '/static/buttons/generic_button_component.html.haml',
+});
+
+genericButtonController.$inject = ['miqService', '$window'];
+
+function genericButtonController(miqService) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.saveable = miqService.saveable;
+
+    vm.resetClicked = function() {
```

Comment:
The `miqAjaxButton` implementation is a complete no-no. No need for ajax really just to get to a specific url. Currently it's confined to only the older forms.
The new design should just do simple redirects (like this one)
As we convert other controllers to components, we will get that uniformity for the `cancel` operation.